test/unit/destructuring.spec.ts

```diff
@@ -78,6 +91,16 @@ test.each(assignmentTestCases)("in assignment expression (%p)", ({ binding, valu
     `.expectToMatchJsResult();
 });
 
+test("destructuring in loop", () => {
+    util.testFunction`
+        let result = 0;
+        for (const [[x]] of [[[1]]]) {
+            result = x;
+        }
+        return result;
+        `.expectToMatchJsResult();
+});
```

Comment:
We already have some destructuring in `for...of` loop tests in `loops.spec.ts`.
Alternatively we might just run all of test cases there (filtering out currently unsupported things):

```ts
test.each(testCases)("in for...of array (%p)", ({ binding, value }) => {
    util.testFunction`
        let ${allBindings};
        {
            for (const ${binding} of [${value}]) {
                return { ${allBindings} };
            }
        }
    `.expectToMatchJsResult();
});
```

Reply:
moved